### PR TITLE
Added pre-release changelog to release script

### LINKED ISSUE #746: Pre-releases should not generate a changelog
When releasing with the `--pre` option we should not generate a changelog. This may require a deviation from the current changelog tool to pick up changes since the last non-pre-release release.

### PR DESCRIPTION
**Pull request by mtscout6**
Closes #746

Creates a different changelog when releasing a pre-release, then when an actual release occurs it is removed.

The idea is that the actual release would contain the same release information but in a single entry.

### CODE REVIEW

File: tools/release-scripts/release.js [hunk 1]
```diff
@@ -42,6 +42,10 @@ const yargsConf = yargs
 const argv = yargsConf.argv;
 setExecOptions(argv);
 
+if (argv.dryRun) {
+  console.log('DRY RUN'.magenta);
```
Comment (mathieumg) [inline tools/release-scripts/release.js]: I'm just curious, where is `console` overridden to support these colors?

**Comment by mathieumg** [comment]
LGTM.

File: tools/release-scripts/release.js (hunk 1 above)
Comment (dozoisch) [inline tools/release-scripts/release.js]: `import 'colors'`

Comment (mathieumg) [inline tools/release-scripts/release.js]: I was looking for it in this file, thanks!

**Comment by mtscout6** [comment]
It's not quite ready to merge I found a kink in how it works. I'm working on it now and should have it wrapped up later today.

File: tools/release-scripts/release.js (hunk 1 above)
Comment (mtscout6) [inline tools/release-scripts/release.js]: @mathieumg You're right I needed to add that to this file as well, good catch.

**Comment by mtscout6** [comment]
Should be good to go now.

**Comment by mtscout6** [comment]
@react-bootstrap/collaborators Any other concerns with this?

**Comment by AlexKVal** [comment]
:shipit:

**Pull request merged**